src/main/java/org/jenkins_ci/plugins/run_condition/core/StringsMatchCondition.java

```diff
@@ -32,17 +32,26 @@
 import org.jenkinsci.plugins.tokenmacro.TokenMacro;
 import org.kohsuke.stapler.DataBoundConstructor;
 
+import java.util.Map;
+import java.util.logging.Logger;
+
 public class StringsMatchCondition extends AlwaysPrebuildRunCondition {
 
     final String arg1;
     final String arg2;
     final boolean ignoreCase;
+    final boolean shellVariables;
 
     @DataBoundConstructor
-    public StringsMatchCondition(final String arg1, final String arg2, final boolean ignoreCase) {
+    public StringsMatchCondition(final String arg1, final String arg2, final boolean ignoreCase,final boolean shellVariables) {
```

Comment:
Public APIs in Jenkins are usually treated as stable so that we don't break compilation of consumers of the public API.  In this case, rather than adding the argument to the `DataBoundConstructor`, I think that you want to define a DataBoundSetter that can define the value after the object has been created. 
```suggestion
    public StringsMatchCondition(final String arg1, final String arg2) {
```

Reply:
@MarkEWaite Done sir